Sources/GitBuddyCore/ChangelogProducer.swift

```diff
@@ -10,7 +10,31 @@ import Foundation
 import OctoKit
 import SPMUtility
 
-public final class GitBuddy {
+struct ChangelogCommand: Command {
+    let command = "changelog"
+    let description = "Create a changelog for GitHub repositories"
+    let sinceTag: OptionArgument<String>
+    let baseBranch: OptionArgument<String>
+    let verbose: OptionArgument<Bool>
+
+    init(parser: ArgumentParser) {
+        let subparser = parser.add(subparser: command, overview: description)
+        sinceTag = subparser.add(option: "--sinceTag", shortName: "-s", kind: String.self, usage: "The tag to use as a base")
+        baseBranch = subparser.add(option: "--baseBranch", shortName: "-b", kind: String.self, usage: "The base branch to compare with")
+        verbose = subparser.add(option: "--verbose", kind: Bool.self, usage: "Show extra logging for debugging purposes")
+    }
+
+    @discardableResult func run(using arguments: ArgumentParser.Result, environment: [String: String]) throws -> String {
+        let changelogProducer = try ChangelogProducer(sinceTag: arguments.get(sinceTag),
+                                                      baseBranch: arguments.get(baseBranch),
+                                                      verbose: arguments.get(verbose) ?? false,
+                                                      environment: environment)
+        return try changelogProducer.run()
+    }
+}
+
+/// Capable of producing a changelog based on input parameters.
+final class ChangelogProducer: URLSessionInjectable {
```

Comment:
It took me a while to figure out where the `urlSession` property came from. Wouldn't it be more straightforward to inject the dependency in t he initializer in combination with a default argument? Let me know your thoughts.

Reply:
It's taken from the way we work in Coyote as well. Adding it to the init creates a lot of clutter while it's only been used for testing. I would like to keep it as it is, just like we do in Coyote

Reply:
Having had a look at the next PRs it makes sense to me to use this pattern as we configure the global session once and then use it the same way everywhere else.